cmd/zfs/zfs_main.c

```diff
@@ -6787,12 +6807,12 @@ get_one_dataset(zfs_handle_t *zhp, void *data)
 }
 
 static void
-get_all_datasets(get_all_cb_t *cbp, boolean_t verbose, const char *dataset)
+get_all_datasets(get_all_cb_t *cbp, boolean_t verbose, const char *filesystem)
 {
 	get_all_state_t state = {
 	    .ga_verbose = verbose,
 	    .ga_cbp = cbp,
-	    .ga_dataset = dataset
+	    .ga_filesystem = filesystem
```

Comment:
As in the struct, `filesystem` here really ought to be `parent` to make it clear you're looking for datasets with a give *parent*.